core/trace/src/main/scala/org/typelevel/otel4s/trace/SpanContext.scala

```diff
@@ -91,3 +91,14 @@ object SpanContext {
   def fromContext[F[_]](context: Vault): Option[SpanContext] =
     context.lookup(key)
 }
+
+private final case class SpanContextImpl(
+    traceId: ByteVector,
+    traceIdHex: String,
+    spanId: ByteVector,
+    spanIdHex: String,
+    samplingDecision: SamplingDecision,
+    isValid: Boolean,
+    isRemote: Boolean,
+    storeInContext: Vault
```

Comment:
`SpanContext` defines `storeInContext` as follows:
```scala
def storeInContext(context: Vault): Vault =
  context.insert(SpanContext.key, this)
```

Should we make it `final` and remove it from the `SpanContextImpl`?